src/views/auth/LoginForm.vue

```diff
@@ -27,14 +27,32 @@ const { handleSubmit } = useForm({
   validationSchema: toTypedSchema(UserLoginSchema),
 })
 
+watch(
+  () => userStore.authenticated,
+  (newVal) => {
+    if (newVal) {
+      if ('next' in router.currentRoute.value.query) {
+        router.push(router.currentRoute.value.query.next as string)
+        return
+      }
+      router.replace({ path: '/' })
+    }
+  }
+)
+
 const onSubmit = handleSubmit(async (values) => {
   isLoading.value = true
   authError.value = ''
 
   try {
     const result = await authenticateUser(values)
-    if (!result.isError) {
-      userStore.login()
+    if (result.status != 401) {
```

Comment:
it's better to use `result.status == 200` 
The server can return other status codes

Reply:
updated